src/util/README.md

```diff
@@ -3,25 +3,66 @@
 
 # Folder util
 
-\author Martin Brain, Owen Jones
+\author Martin Brain, Owen Jones, Chris Smowton
 
 \section data_structures Data Structures
 
 \ref util contains some of the key data-structures used in the
 CPROVER codebase.
 
-\subsection irept_section irept: a 4-tuple (data, named-sub, comments, sub)
+\subsection irept_section irept: a general-purpose polymorphic tree
```

Comment:
Hm... `polymorphic`... in some place it should be mentioned that irept derived classes must not add any data members nor virtual methods.